tutorials/tracing.md

```diff
@@ -0,0 +1,182 @@
+# CRI-O Tracing
+
+## Configuration
+
+To enable [OpenTelemetry][otel] tracing support in CRI-O, either start `crio`
+with `--enable-tracing` or add the corresponding option to a config overwrite,
+for example `/etc/crio/crio.conf.d/01-tracing.conf`:
+
+[otel]: https://opentelemetry.io
+
+```toml
+[crio.tracing]
+enable_tracing = true
+```
+
+Traces in CRI-O get exported via the [OpenTelemetry Protocol][otlp] by using an
+[gRPC][grpc] endpoint. This endpoint defaults to `0.0.0.0:4317`, but can be
+configured by using the `--tracing-endpoint` flag or the corresponding TOML
+configuration:
+
+[otlp]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md
+[grpc]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md#otlpgrpc
+
+```toml
+[crio.tracing]
+tracing_endpoint = "0.0.0.0:4317"
+```
+
+The final configuration aspect of OpenTelemetry tracing in CRI-O is the
+`--tracing-sampling-rate-per-million` / `tracing_sampling_rate_per_million`
+configuration, which refers to the amount of samples collected per million
+spans. This means if it being set to `0` (the default), then CRI-O will not
+collect any traces at all. If set to `1000000` (one million), then CRI-O will
+create traces for all created spans.
+
+If CRI-O runs with [conmon-rs][conmon-rs], for example by configuring to use the
+`pod` runtime type like this:
+
+```toml
+[crio.runtime]
+default_runtime = "runc"
+
+[crio.runtime.runtimes.runc]
+runtime_type = "pod"
+```
+
+[conmon-rs]: https://github.com/containers/conmon-rs
+
+Then conmon-rs will export traces and spans in the same way CRI-O does
+automatically. Both CRI-O and conmon-rs will correlate their logs to the trances
+and spans. If the connection to the OTLP instance gets lost, then CRI-O will not
+block, but all the traces during that time will be lost.
+
```

Comment:
Lets mention that logs are saved if `logs` package is used? This is however is more of a developer docs